apps/cli/lib/cli/state_supervisor.ex

```diff
@@ -0,0 +1,18 @@
+defmodule CLI.StateSupervisor do
+  use Supervisor
+  alias CLI.State
+
+  @moduledoc """
+  Supervising CLI State and creating the ETS table that would hold chain changes for JSONRPC.
+  """
+  def start_link(chain) do
+    Supervisor.start_link(__MODULE__, chain)
+  end
+
+  def init(chain) do
+    :cli = :ets.new(:cli, [:set, :public, :named_table])
+    children = [worker(State, [chain])]
```

Comment:
Should we use the new format for specifications? Since `State` is a gen server, I think this would work ->`{State, [chain]}`

Reply:
tuple childs are new?

Reply:
Not extremely new, but they came in [v1.5](https://elixir-lang.org/blog/2017/07/25/elixir-v1-5-0-released/#streamlined-child-specs)

Reply:
aha! was not aware of these new api changes, good point! fixed.